### PR TITLE
[docs] Copyedit `pages.ts` navigation

### PR DESCRIPTION
**Pull request by mapache-salvaje**
related to: https://github.com/mui/mui-x/issues/2904#issuecomment-2383693360

- renamed "Advanced" to "Custom behavior"
- enforced sentence case across all titles
- removed ampersands
- shortened "Custom slots and subcomponents" to "Custom subcomponents"
- removed "Group and Pivot" group to expose more Premium features

### CODE REVIEW

**Review by LukasTy — COMMENTED** [review]
If you feel that `slots and` part doesn't bring any clarity, I'm fine with it. 👌 
But if we rename a menu item, shouldn't we also align that page heading as well? 🤔 

I'd ask for a pair of eyes from @mui/xgrid given that most renaming impacts their links.

**Comment by flaviendelangle** [comment]
> shortened "Custom slots and subcomponents" to "Custom subcomponents"

The main goal was to improve the searchability of this section.
`slots` is the name of the prop and the name of the technical concept.

If we remove it from the sidebar, then I think it should at least be kept on the page title so that Algolia is able to find this page when searching for `slots`

File: docs/data/pages.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ const pages: MuiPage[] = [
       { pathname: '/x/react-data-grid/performance' },
       {
         pathname: '/x/react-data-grid-group-pivot',
-        title: 'Group & Pivot',
+        title: 'Grouping and pivoting',
```
Comment (cherniavskii) [inline docs/data/pages.ts:113]: Side note: What do you think about flattening this arbitrary group?

**Review by cherniavskii — APPROVED** [review]
Thanks!

File: docs/data/pages.ts (hunk 1 above)
Comment (flaviendelangle) [reply docs/data/pages.ts:113]: To give some context, the group was based on AG Grid doc (and existed even before I arrived at MUI and way before any of the feature were implemented).

**Comment by flaviendelangle** [comment]
> As I spend more time with the X nav bar, I'm starting to feel like the Data Grid navigation could use more structure in the form of subheaders to group different categories of docs (for example, AG Grid has "Core Features," "Advanced Features," and "Miscellaneous")

I agree that grouping pages like we are doing on the pickers can be a nice middle ground between a collapsed folder and a totally flat structure.

Just be careful, last time I tried, when you have a group and then a page below, the page below looks like its in the group, we should probably improve the padding to show that its outside of it.

**Pull request merged**